### PR TITLE
Draw adjustments

### PR DESCRIPTION
**Pull request by FilipLeitner**
Mostly changes requested by Petr Vacha

...There have also been a debate about the drawing finishing using right click. Whether ill be more intuitive to preserve last sketch vertex (add it to the feature before finishing drawing)  or to preserve only the 'stable' ones  (ones that are added by left click).

In the PR the first option is used.  On the other hand tester who tested it on v1.24 asked for the second one.

### CODE REVIEW

**Comment by raitisbe** [comment]
> Mostly changes requested by Petr Vacha
> 
> ...There have also been a debate about the drawing finishing using right click. Whether ill be more intuitive to preserve last sketch vertex (add it to the feature before finishing drawing) or to preserve only the 'stable' ones (ones that are added by left click).
> 
> In the PR the first option is used. On the other hand tester who tested it on v1.24 asked for the second one.

Make it configurable :)

File: components/draw/draw.service.ts [hunk 1]
```diff
@@ -375,6 +423,41 @@ export class HsDrawService {
         source: this.source,
         type: /** @type {GeometryType} */ this.type,
         style: changeStyle ? changeStyle() : undefined,
+        condition: (e) => {
```
Comment (DailisLangovskis) [inline components/draw/draw.service.ts:426]: Should check if user tries to right click without adding any points on the map, otherwise getting null value.

**Comment by DailisLangovskis** [comment]
Drawn features are not showing either on the map, or legend. This should have been fixed by #1066 .

File: components/draw/draw.service.ts (hunk 1 above)
Comment (FilipLeitner) [reply components/draw/draw.service.ts:426]: Not sure I understand what do you mean.

**Comment by FilipLeitner** [comment]
> Drawn features are not showing either on the map, or legend. This should have been fixed by #1066 .

Could you be more specific please. Not observing such a behvior
image

File: components/draw/draw.service.ts (hunk 1 above)
Comment (DailisLangovskis) [reply components/draw/draw.service.ts:426]: When I select line or polygon, and click right mouse click on the map, I get error, cannot read length of undefined.

**Comment by DailisLangovskis** [comment]
> > Drawn features are not showing either on the map, or legend. This should have been fixed by #1066 .
> 
> Could you be more specific please. Not observing such a behvior
> image

Yes, this was on my side, because I can replicate it again :D

File: components/draw/draw.service.ts (hunk 1 above)
Comment (DailisLangovskis) [reply components/draw/draw.service.ts:426]: https://drive.google.com/file/d/1-g6Kb_CRPaqOcBa6lDvIpczOVP2I9idM/view This is the error I was experiencing.

Comment (FilipLeitner) [reply components/draw/draw.service.ts:426]: Oh I see, right . Thanks.

**Review by DailisLangovskis — APPROVED** [review]
Everything seems to be ok now!

File: components/draw/draw.service.ts [hunk 2]
```diff
@@ -171,7 +184,7 @@ export class HsDrawService {
         visible: true,
         removable: true,
         editable: true,
-        synchronize: true,
```
Comment (raitisbe) [inline components/draw/draw.service.ts:174]: What if we set this to true if and only if we have layman endpoint defined?

File: components/draw/draw.service.ts [hunk 3]
```diff
@@ -335,12 +348,47 @@ export class HsDrawService {
...
+      HsConfirmDialogComponent,
+      {
+        message: this.TranslateService.instant('Really delete this layer?'),
+        title: this.TranslateService.instant('Confirm delete'),
+      }
+    );
+    const confirmed = await dialog.waitResult();
+    if (confirmed == 'yes') {
+      this.HsMapService.map.removeLayer(this.selectedLayer);
+      if (this.selectedLayer.get('synchronize') == true) {
```
Comment (raitisbe) [inline components/draw/draw.service.ts:367]: Would be better to move insides of this if condition to layman service to seperate concerns. Drawing service shouldnt manage layers and do synchronization to endpoints.

File: components/draw/draw.service.ts [hunk 4]
```diff
@@ -375,6 +423,41 @@ export class HsDrawService {
...
+          if (e.originalEvent.buttons === 1) {
+            //left click
+            return true;
+          } else if (e.originalEvent.buttons === 2) {
+            //right click
+            const minPoints = this.HsConfig.preserveLastSketchPoint ? 1 : 0;
+            if (this.type == 'Polygon') {
+              const vertexCount = this.draw.sketchLineCoords_?.length;
+              if (vertexCount >= 4 - minPoints) {
+                setTimeout(() => {
```
Comment (raitisbe) [inline components/draw/draw.service.ts:436]: Try refactoring the duplicate 
```
 setTimeout(() => {
                  if (minPoints == 0) {
                    this.removeLastPoint();
                  }
                  this.draw.finishDrawing();
                }, 250);
``` 
into seperate function so it's more readable.

**Review by raitisbe — CHANGES_REQUESTED** [review]

File: components/draw/draw.service.ts (hunk 4 above)
Comment (FilipLeitner) [reply components/draw/draw.service.ts]: 29c25580f13c56e014bb0b95065009ba68ec8403

File: components/draw/draw.service.ts (hunk 3 above)
Comment (FilipLeitner) [reply components/draw/draw.service.ts:365]: 15564b091777044edb8d270dce08ca85724d0e90

File: components/draw/draw.service.ts (hunk 2 above)
Comment (FilipLeitner) [reply components/draw/draw.service.ts:174]: 7cc8eaf00c19caaad4a9bb9d5643cb06ffa1d422

**Review by raitisbe — APPROVED** [review]

**Pull request merged**